src/Components/Components/src/Virtualization/IVirtualizationService.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// Describes services enabling platform-specific virtualization.
+    /// </summary>
+    public interface IVirtualizationService
```

Comment:
While I recognize some benefits of the `IVirtualizationService` design concept, I'm not yet 100% sold on it :) It seems to be based on the following:

 * All rendering platforms can be assumed to provide an implementation for it
 * There are use cases for sharing the exact same `<Virtualization>` component type across multiple rendering platforms

I'm not sure either of those are necessarily true. At least, I haven't yet thought of a case where someone would be building a component that's high-level enough to have UI opinions like "I should be virtualized" while also low-level enough to be applicable across multiple rendering platforms.

The second concern I have is that the abstraction hasn't exactly worked out without leaking. The `VirtualizeBase` code embeds some web-specific concepts (i.e., the notion of a `div` and a CSS `style` attribute). If we were keen to avoid this abstraction leakage, would that potentially work out most easily by keeping `VirtualizeBase` in `M.A.Components` and then implementing a `Virtualize` in `M.A.Components.Web` that can directly do JS interop without having to go through an `IVirtualizationService`?

For the abstraction to be as general as possible, it's worth considering whether the whole concept of "spacer" elements could be limited to the `.Web` implementation. Other UI platforms might have a native implementation of virtualization that can work more conveniently without needing any spacer things, and in fact might not even be able to have any equivalent to a spacer depending on their rules about allowed UI widget hierarchies.

If the end result is that there is very little non-web-specific stuff left, then arguably we might have taken a misstep by thinking there could be a non-platform-specific version virtualization (which would be my mistake in the original suggestion). But TBH I don't know, maybe there *is* enough to make the platform-independent base abstraction remain worthwhile!

Reply:
I agree that it would be nice to remove `IVirtualizationService` if possible. I've gone ahead and committed a different approach that combines `Virtualize` and `InfiniteScroll`, which removes most of the need for a service to provide shared functionality across components. I did end up leaving in `VirtualizeBase`, as I do believe there are still a few concepts that can be generalized across platforms.